cryptography/src/lib.rs

```diff
@@ -124,6 +124,42 @@ pub trait Hasher: Clone + Send + Sync + 'static {
     fn random<R: Rng + CryptoRng>(rng: &mut R) -> Digest;
 }
 
+pub trait BatchScheme {
+    /// Create a new Batch
+    fn new() -> Self;
+
+    /// Append item to the batch
+    ///
+    /// The message should not be hashed prior to calling this function. If a particular scheme
+    /// requires a payload to be hashed before it is signed, it will be done internally.
+    ///
+    /// A namespace should be used to prevent replay attacks. It will be prepended to the message so
+    /// that a signature meant for one context cannot be used unexpectedly in another (i.e. signing
+    /// a message on the network layer can't accidentally spend funds on the execution layer). See
+    /// [union_unique](commonware_utils::union_unique) for details
+    fn add(
+        &mut self,
+        namespace: Option<&[u8]>,
+        message: &[u8],
+        public_key: &PublicKey,
+        signature: &Signature,
+    ) -> Result<(), BatchError>;
+
+    /// Perform batch verification over the previously added items.
+    fn verify<R: RngCore + CryptoRng>(self, rng: R) -> bool;
```

Comment:
Batched BLS signature verification (where each `(public_key, message)` may be unique) also requires using randomness, so this interface looks very reasonable (defends against the same "class" of errors).

https://docs.rs/blst/0.3.13/blst/min_pk/struct.Signature.html#method.verify_multiple_aggregate_signatures